openslides_backend/action/actions/meeting/archive.py

```diff
@@ -22,6 +23,41 @@ class MeetingArchive(UpdateAction, GetMeetingIdFromIdMixin):
     model = Meeting()
     schema = DefaultSchema(Meeting()).get_update_schema()
 
+    def validate_instance(self, instance: dict[str, Any]) -> None:
+        super().validate_instance(instance)
+
+        messages = []
+
+        active_speaker_exists = self.datastore.exists(
+            "speaker",
+            And(
+                FilterOperator("meeting_id", "=", instance["id"]),
+                FilterOperator("begin_time", "!=", None),
+                FilterOperator("end_time", "=", None),
+            ),
+            lock_result=False,
+        )
+        active_polls = self.datastore.filter(
+            "poll",
+            And(
+                FilterOperator("meeting_id", "=", instance["id"]),
+                FilterOperator("state", "=", "started"),
+            ),
+            ["title"],
+            lock_result=False,
+        )
```

Comment:
No need to treat speakers and polls differently, just use `datastore.exists` for both